src/components/tag-box/tag-box.js

```diff
@@ -0,0 +1,367 @@
+import { isElement } from '../../main';
+
+const hiddenNumberWidth = 32;
+const iconToggleWidth = 24;
+const gapSpace = 8;
+const lastItemWidth = hiddenNumberWidth + gapSpace + iconToggleWidth;
+
+function createTagBoxItem(content) {
+  const template = document.createElement('template');
+  template.innerHTML = `
+  <cdg-tag-box-item>${content}</cdg-tag-box-item>
+  `;
+  return template;
+}
+
+export class CdgTagBoxContainer extends HTMLElement {
+  labelElement;
+  containerElement;
+  mainElement;
+  bottomElement;
+  helperTextElement;
+  hiddenItemElements = [];
+  hiddenNumberElement;
+  ctaButtons;
+  toggleIconELement;
+  inputElement;
+  _show = false;
+  _type = 'text-box';
+
+  get type() {
+    return this._type;
+  }
+
+  set type(value) {
+    this._type = value;
+    this.containerElement.classList.add(`cdg-tag-box-${this._type}`);
+  }
+
+  static get observedAttributes() {
+    return ['type'];
+  }
+
+  constructor() {
+    super();
+    this.mainElement = document.createElement('div');
+    this.mainElement.classList.add('cdg-tag-box-main');
+
+    if (this.hasAttribute('label-left')) {
+      this.classList.add('cdg-tag-box-label-left');
+    }
+
+    this.createInputContent();
+
+    if (this.hasAttribute('label')) {
+      this.labelElement = document.createElement('span');
+      this.labelElement.classList.add('cdg-input-label');
+      this.labelElement.innerHTML = this.getAttribute('label');
+      this.prepend(this.labelElement);
+    }
+
+    this.createBottomContainer();
+  }
+
+  createInputContent() {
+    this.containerElement = document.createElement('div');
+    this.containerElement.classList.add('cdg-tag-box-container');
+    this.containerElement.tabIndex = 0;
+
+    if (this.hasAttribute('inputable')) {
+      this.inputElement = document.createElement('input');
+      this.inputElement.classList.add('cdg-tag-box-input-text');
+      this.containerElement.addEventListener(
+        'focus',
+        this.handleContainerFocus.bind(this)
+      );
+      this.containerElement.addEventListener(
+        'blur',
+        this.handleContainerBlur.bind(this)
+      );
+
+      this.inputElement.addEventListener(
+        'blur',
+        this.handleInputTextBlur.bind(this)
+      );
+
+      this.inputElement.addEventListener(
+        'keyup',
+        this.handleInputTextKeyUp.bind(this)
+      );
+    }
+
+    this.type = this.getAttribute('type') || 'text-box';
```

Comment:
Convert this to

get type () {

return this.getAttribute('type') || 'text-box';

}